configure.ac

```diff
@@ -473,61 +483,28 @@ AS_IF([test "x$enable_bow_docs" = "xyes"], [
 win32_target_dll=0
 COMPILER_IS_GCC=0
 
-AS_IF([test "x$ac_cv_c_compiler_gnu" = "xyes"], [
-	MN_ADD_CFLAGS(-std=gnu99)
-
-	MN_GCC_VERSION
-
-	AS_IF([test "x$GCC_MAJOR_VERSION$GCC_MINOR_VERSION" = "x42"], [
-		AC_MSG_WARN([****************************************************************])
-		AC_MSG_WARN([** GCC version 4.2 warns about the inline keyword for no good **])
-		AC_MSG_WARN([** reason but the maintainers do not see it as a bug.         **])
-		AC_MSG_WARN([** See http://gcc.gnu.org/bugzilla/show_bug.cgi?id=33995      **])
-		AC_MSG_WARN([** Using -fgnu-inline to avoid this stupidity.                **])
-		AC_MSG_WARN([****************************************************************])
-		MN_ADD_CFLAGS([-fgnu89-inline])
-	])
-
-	CFLAGS="$CFLAGS -Wall"
-	CXXFLAGS="$CXXFLAGS -Wall"
-
-	MN_ADD_CFLAGS([-Wextra])
-	MN_ADD_CFLAGS([-Wdeclaration-after-statement])
-	MN_ADD_CFLAGS([-Wpointer-arith])
-	MN_ADD_CFLAGS([-Wvla])
-
-	AC_LANG_PUSH([C++])
-	MN_ADD_CXXFLAGS([-Wextra])
-	MN_ADD_CXXFLAGS([-Wpointer-arith])
-	AC_LANG_POP([C++])
-
-	AS_IF([test "x$enable_stack_smash_protection" = "xyes"], [
-		XIPH_GCC_STACK_PROTECTOR
-		XIPH_GXX_STACK_PROTECTOR
-	])
+common_flags="-Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wshadow -Wwrite-strings -Wundef -Wuninitialized -Winit-self"
+AX_APPEND_COMPILE_FLAGS([${common_flags} -Wdeclaration-after-statement -Wvla -Wbad-function-cast -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Waggregate-return], [CFLAGS])
 
-	AS_IF([test "x$enable_test_coverage" = "xyes"], [
-		# MN_ADD_CFLAGS([-ftest-coverage])
-		MN_ADD_CFLAGS([-coverage])
-	])
+AC_LANG_PUSH([C++])
+AX_APPEND_COMPILE_FLAGS([${common_flags} -Wctor-dtor-privacy -Wnon-virtual-dtor -Woverloaded-virtual -Wreorder -Wsign-promo], [CXXFLAGS])
+AC_LANG_POP([C++])
 
-	dnl some distributions (such as Gentoo) have _FORTIFY_SOURCE always
-	dnl enabled. We test for this situation in order to prevent polluting
-	dnl the console with messages of macro redefinitions.
-	AX_ADD_FORTIFY_SOURCE
-
-	common_flags="-Wcast-align -Wcast-qual -Wshadow -Wwrite-strings -Wundef -Wuninitialized -Winit-self"
+AS_IF([test "x$enable_stack_smash_protection" = "xyes"], [
+	XIPH_GCC_STACK_PROTECTOR
+	XIPH_GXX_STACK_PROTECTOR
+])
 
-	# -Winline -Wconversion "
-	CFLAGS="$CFLAGS     $common_flags -Wbad-function-cast -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Waggregate-return"
-	CXXFLAGS="$CXXFLAGS $common_flags -Wctor-dtor-privacy -Wnon-virtual-dtor -Woverloaded-virtual -Wreorder -Wsign-promo"
+AS_IF([test "x$enable_test_coverage" = "xyes"], [
+	AX_APPEND_COMPILE_FLAGS([-coverage], [CFLAGS])
+])
```

Comment:
There is an existing indentation scheme in this file and in this project that should be adhered to. Eg:
```
AS_IF([test "x$enable_test_coverage" = "xyes"], [
	AX_APPEND_COMPILE_FLAGS([-coverage], [CFLAGS])
	])
```